radicale/__init__.py

```diff
@@ -92,6 +93,14 @@ def __init__(self, address, handler):
         self.server_activate()
 
 
+class ThreadedHTTPServer(socketserver.ThreadingMixIn, HTTPServer):
```

Comment:
Atomic file writes are not sufficient to solve the issue of partial uploads, because changes of multiple files might be needed to transfer from one valid state to another (e.g. moving of whole collections).
Solutions to this problem are quite  elaborate. If it's a requirement, just use a database that supports ACID.